### ISSUE TITLE
Ability to add nesting in buckets

### ISSUE THREAD

**Issue by scalvert** [kaliber5/ember-cli-bundlesize#2]
Our app has a TON of engines. Each of those engines, which are mostly lazy engines, produce an individual engine.js, engine-vendor.js, and engine.css file. It would be great to be able to group each under a config bucket, which would represent the options for that isolated engine.

```js
module.exports = {
  'engine-1': {
    javascript: {
      pattern: 'assets/*.js',
      limit: '500KB',
      compression: 'gzip'
    },
    css: {
      pattern: 'assets/*.css',
      limit: '50KB',
      compression: 'gzip'
    }
  },
  'engine-2': {
    javascript: {
      pattern: 'assets/*.js',
      limit: '500KB',
      compression: 'gzip'
    },
    css: {
      pattern: 'assets/*.css',
      limit: '50KB',
      compression: 'gzip'
    }
  }
};
```

**Comment by scalvert** [comment]
I wonder if it's worth modifying the config such that, by default, the config assumes buckets are keyed by name. This would allow the default config to have an `app` bucket by default, but would operate against _n_ buckets in that config. 

Thoughts?

**Comment by simonihmig** [comment]
@scalvert thanks for your feedback! I just wonder, what benefit would your example above give you compared to something like the following?

```js
module.exports = {
  'engine-1-javascript': {
    pattern: 'assets/*.js',
    limit: '500KB',
    compression: 'gzip'
  },
  'engine-1-css': {
    pattern: 'assets/*.css',
    limit: '50KB',
    compression: 'gzip'
  }
 ...
};
```

Currently the key is only used as a name for the bucket when running the test command, so not sure what benefit nesting would provide here?

**Comment by scalvert** [comment]
Yep, I'm totally aware we can have a flat config. I guess I'm thinking of a case where you have ~60 engines, each with their own .js and .css. That flat list becomes hard to scan...

I realize I'm on the extreme here.

**Comment by simonihmig** [comment]
> I realize I'm on the extreme here.

Probably yes, but I would love this tool to be also of value for apps of that scale. 

Not sure though when I will find the time for this. If you or someone else is up for a PR, that would be awesome of course!

**Comment by simonihmig** [comment]
Closed by #7 :tada:

**Comment by scalvert** [comment]
🎉
